src/pages/home/report/ReportActionItem.js

```diff
@@ -74,144 +73,132 @@ const propTypes = {
 
 const defaultProps = {
     draftMessage: '',
-    hasOutstandingIOU: false,
     preferredSkinTone: CONST.EMOJI_DEFAULT_SKIN_TONE,
 };
 
-class ReportActionItem extends Component {
-    constructor(props) {
-        super(props);
-        this.popoverAnchor = undefined;
-        this.state = {
-            isContextMenuActive: ReportActionContextMenu.isActiveReportAction(props.action.reportActionID),
-        };
-        this.checkIfContextMenuActive = this.checkIfContextMenuActive.bind(this);
-        this.showPopover = this.showPopover.bind(this);
-        this.renderItemContent = this.renderItemContent.bind(this);
-        this.toggleReaction = this.toggleReaction.bind(this);
-    }
-
-    shouldComponentUpdate(nextProps, nextState) {
-        return this.props.displayAsGroup !== nextProps.displayAsGroup
-            || this.props.draftMessage !== nextProps.draftMessage
-            || this.props.isMostRecentIOUReportAction !== nextProps.isMostRecentIOUReportAction
-            || this.props.hasOutstandingIOU !== nextProps.hasOutstandingIOU
-            || this.props.shouldDisplayNewMarker !== nextProps.shouldDisplayNewMarker
-            || !_.isEqual(this.props.action, nextProps.action)
-            || this.state.isContextMenuActive !== nextState.isContextMenuActive;
-    }
+function ReportActionItem(props) {
+    const [isContextMenuActive, setIsContextMenuActive] = useState(ReportActionContextMenu.isActiveReportAction(props.action.reportActionID));
+    const previousDraftMessageRef = useRef('');
```

Comment:
Does this component ever do an initial render with a value for `draftMessage`? If so, maybe this should be initialized with `props.draftMessage`?